test/Microsoft.AspNet.Mvc.Core.Test/HttpResponseStreamWriterTest.cs

```diff
@@ -0,0 +1,337 @@
+
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class HttpResponseStreamWriterTest
+    {
+        [Fact]
+        public async Task DoesNotWriteBOM()
+        {
+            // Arrange
+            var memoryStream = new MemoryStream();
+            var encodingWithBOM = new UTF8Encoding(encoderShouldEmitUTF8Identifier: true);
+            var writer = new HttpResponseStreamWriter(memoryStream, encodingWithBOM);
+            var expectedData = new byte[] { 97, 98, 99, 100 }; // without BOM
+
+            // Act
+            using (writer)
+            {
+                await writer.WriteAsync("abcd");
+            }
+
+            // Assert
+            Assert.Equal(expectedData, memoryStream.ToArray());
+        }
+
+        [Fact]
+        public async Task DoesNotFlush_UnderlyingStream_OnClosingWriter()
+        {
+            // Arrange
+            var stream = new TestMemoryStream();
+            var writer = new HttpResponseStreamWriter(stream, Encoding.UTF8);
+
+            // Act
+            await writer.WriteAsync("Hello");
+            writer.Close();
+
+            // Assert
+            Assert.Equal(0, stream.FlushCallCount);
+            Assert.Equal(0, stream.FlushAsyncCallCount);
+        }
+
+        [Fact]
+        public async Task DoesNotFlush_UnderlyingStream_OnDisposingWriter()
+        {
+            // Arrange
+            var stream = new TestMemoryStream();
+            var writer = new HttpResponseStreamWriter(stream, Encoding.UTF8);
+
+            // Act
+            await writer.WriteAsync("Hello");
+            writer.Dispose();
+
+            // Assert
+            Assert.Equal(0, stream.FlushCallCount);
+            Assert.Equal(0, stream.FlushAsyncCallCount);
+        }
+
+        [Fact]
+        public async Task DoesNotClose_UnderlyingStream_OnDisposingWriter()
+        {
+            // Arrange
+            var stream = new TestMemoryStream();
+            var writer = new HttpResponseStreamWriter(stream, Encoding.UTF8);
+
+            // Act
+            await writer.WriteAsync("Hello");
+            writer.Close();
+
+            // Assert
+            Assert.Equal(0, stream.CloseCallCount);
+        }
+
+        [Fact]
+        public async Task DoesNotDispose_UnderlyingStream_OnDisposingWriter()
+        {
+            // Arrange
+            var stream = new TestMemoryStream();
+            var writer = new HttpResponseStreamWriter(stream, Encoding.UTF8);
+
+            // Act
+            await writer.WriteAsync("Hello world");
+            writer.Dispose();
+
+            // Assert
+            Assert.Equal(0, stream.DisposeCallCount);
+        }
+
+        [Theory]
+        [InlineData(1023)]
+        [InlineData(1024)]
+        [InlineData(1050)]
+        [InlineData(2048)]
+        public async Task FlushesBuffer_OnClose(int byteLength)
```

Comment:
yeah, i thought about it...in many of the tests here we verify the stream length(which is in bytes) against this length...and the char length here matches with the byte length as I am using only single byte characters like 'a'..